### ISSUE TITLE
invalid-name check for TypeVar should allow for digits in names

### ISSUE THREAD

**Issue by sodul** [pylint-dev/pylint#8499]
### Bug description

```python
"""Ec2T example."""

from typing import TypeVar

T = TypeVar('T')
EcTwoT = TypeVar('EcTwoT')
Ec2T = TypeVar('Ec2T')
```

### Command used

```shell
pylint typevar.py
```

### Pylint output

```shell
************* Module typevar
typevar.py:7:0: C0103: Type variable name "Ec2T" doesn't conform to predefined naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 5.00/10, +2.50)
```

### Expected behavior

`Ec2T` should be valid.

### Pylint version

```shell
pylint 2.17.0
astroid 2.15.0
Python 3.11.2 (main, Mar 20 2023, 21:17:15) [Clang 14.0.0 (clang-1400.0.29.202)]
```

### OS / Environment

MacOS 13.2.1

**Comment by sodul** [comment]
Relates to #8485
